diff --git a/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c b/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c
index b3e02f6..97b53b0 100644
--- a/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c
+++ b/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c
@@ -171,8 +171,9 @@ static boolean do_winsys_init(struct radeon_drm_winsys *ws)
 
     /* Get DRM version. */
     version = drmGetVersion(ws->fd);
-    if (version->version_major != 2 ||
-        version->version_minor < 3) {
+    if (version->version_major != 3 && (
+		version->version_major != 2 ||
+		version->version_minor < 3)) {
         fprintf(stderr, "%s: DRM version is %d.%d.%d but this driver is "
                 "only compatible with 2.3.x (kernel 2.6.34) or later.\n",
                 __FUNCTION__,
