diff --git a/Makefile b/Makefile
index 7575012..11482fa 100644
--- a/Makefile
+++ b/Makefile
@@ -59,7 +59,7 @@ DIRPERM  = 0755
 #SHELLDIR = $(PREFIX)/games
 
 # per discussion in Install.X11 and Install.Qt
-#VARDATND = 
+VARDATND =  x11tiles pet_mark.xbm rip.xpm
 # VARDATND = x11tiles NetHack.ad pet_mark.xbm pilemark.xpm
 # VARDATND = x11tiles NetHack.ad pet_mark.xbm pilemark.xpm rip.xpm
 # for Atari/Gem
diff --git a/src/Makefile b/src/Makefile
index 7da32c7..995b353 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -236,8 +236,8 @@ WINBEOBJ =
 
 #
 #
-#WINSRC = $(WINTTYSRC)
-#WINOBJ = $(WINTTYOBJ)
+WINSRC = $(WINTTYSRC) $(WINX11SRC)
+WINOBJ = $(WINTTYOBJ) $(WINX11OBJ)
 
 # on some systems the termcap library is in -ltermcap or -lcurses
 # on 386 Xenix, the -ltermlib tputs() seems not to work; use -lcurses instead
@@ -257,7 +257,7 @@ WINTTYLIB = -lncurses -ltinfo
 #
 # libraries for X11
 # If USE_XPM is defined in config.h, you will also need -lXpm here.
-WINX11LIB = -lXaw -lXmu -lXext -lXt -lX11
+WINX11LIB = -lXaw -lXmu -lXext -lXt -lX11 -lXpm
 # WINX11LIB = -lXaw -lXmu -lXt -lX11
 # WINX11LIB = -lXaw -lXmu -lXext -lXt -lXpm -lX11 -lm
 # WINX11LIB = -lXaw -lXmu -lXpm -lXext -lXt -lX11 -lSM -lICE -lm # BSD/OS 2.0
@@ -277,7 +277,7 @@ WINGEMLIB = -le_gem -lgem
 # libraries for BeOS 
 WINBELIB = -lbe
 
-#WINLIB = $(WINTTYLIB)
+WINLIB = $(WINTTYLIB) $(WINX11LIB)
 
 # any other strange libraries your system needs (for Sysunix only -- the more
 # specialized targets should already be right)
