diff --git a/include/config.h b/include/config.h
index 141a684..b1bf340 100644
--- a/include/config.h
+++ b/include/config.h
@@ -250,8 +250,8 @@
 
 #if defined(UNIX) && !defined(ZLIB_COMP) && !defined(COMPRESS)
 /* path and file name extension for compression program */
-#define COMPRESS "/usr/bin/compress" /* Lempel-Ziv compression */
-#define COMPRESS_EXTENSION ".Z"      /* compress's extension */
+#define COMPRESS "/usr/bin/xz" /* Lempel-Ziv compression */
+#define COMPRESS_EXTENSION ".xz"      /* compress's extension */
 /* An example of one alternative you might want to use: */
 /* #define COMPRESS "/usr/local/bin/gzip" */ /* FSF gzip compression */
 /* #define COMPRESS_EXTENSION ".gz" */       /* normal gzip extension */
@@ -298,7 +298,7 @@
  *      a tar-like file, thus making a neater installation.  See *conf.h
  *      for detailed configuration.
  */
-/* #define DLB */ /* not supported on all platforms */
+#define DLB 1  /* not supported on all platforms */
 
 /*
  *      Defining INSURANCE slows down level changes, but allows games that
@@ -317,7 +317,7 @@
  * otherwise it will be the current directory.
  */
 #ifndef HACKDIR
-#define HACKDIR "/usr/games/lib/nethackdir"
+#define HACKDIR "GENTOO_HACKDIR"
 #endif
 
 /*
@@ -412,7 +412,7 @@ typedef unsigned char uchar;
  * functions that have been macroized.
  */
 
-/* #define VISION_TABLES */ /* use vision tables generated at compile time */
+#define VISION_TABLES 1 /* use vision tables generated at compile time */
 #ifndef VISION_TABLES
 #ifndef NO_MACRO_CPATH
 #define MACRO_CPATH /* use clear_path macros instead of functions */
diff --git a/include/unixconf.h b/include/unixconf.h
index f63c0bb..637b9bf 100644
--- a/include/unixconf.h
+++ b/include/unixconf.h
@@ -36,7 +36,7 @@
 #define NETWORK        /* if running on a networked system */
                        /* e.g. Suns sharing a playground through NFS */
 /* #define SUNOS4 */   /* SunOS 4.x */
-/* #define LINUX */    /* Another Unix clone */
+#define LINUX          /* Another Unix clone */
 /* #define CYGWIN32 */ /* Unix on Win32 -- use with case sensitive defines */
 /* #define GENIX */    /* Yet Another Unix Clone */
 /* #define HISX */     /* Bull Unix for XPS Machines */
@@ -128,7 +128,7 @@
  * "extra output" method is used, but not all systems provide access to
  * a fine-grained timer.
  */
-/* #define TIMED_DELAY */ /* usleep() */
+#define TIMED_DELAY 1
 #endif
 
 /*
diff --git a/src/Makefile b/src/Makefile
index 9c856bf..fbf59cf 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -253,7 +253,7 @@ WINBEOBJ =
 # WINTTYLIB = -lcurses
 # WINTTYLIB = -lcurses16
 # WINTTYLIB = -lncurses
-#WINTTYLIB = -ltermlib
+WINTTYLIB = -lncurses
 #
 # libraries for X11
 # If USE_XPM is defined in config.h, you will also need -lXpm here.
