diff --git a/include/unixconf.h b/include/unixconf.h
index 0447b82..f63c0bb 100644
--- a/include/unixconf.h
+++ b/include/unixconf.h
@@ -95,13 +95,13 @@
  */
 
 /* #define NO_FILE_LINKS */                       /* if no hard links */
-/* #define LOCKDIR "/usr/games/lib/nethackdir" */ /* where to put locks */
+#define LOCKDIR "GENTOO_STATEDIR"
 
 /*
  * If you want the static parts of your playground on a read-only file
  * system, define VAR_PLAYGROUND to be where the variable parts are kept.
  */
-/* #define VAR_PLAYGROUND "/var/lib/games/nethack" */
+#define VAR_PLAYGROUND "GENTOO_STATEDIR"
 
 /*
  * Define DEF_PAGER as your default pager, e.g. "/bin/cat" or "/usr/ucb/more"
